MigrationV3V4/CSharp/Skills/README.md

```diff
@@ -0,0 +1,82 @@
+# Bot Framework V3 Skills
+
+This sample was created using `Microsoft.Bot.Builder` [v3.30.0](https://www.nuget.org/packages/Microsoft.Bot.Builder/3.30.0) and [v4.7.2](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.7.2) sdks. It shows how to create a simple skill consumer (SimpleRootBot) which sends message activities to any of three V3 bots converted to skills.
+
+![Skill Options](media/SkillOptions.png)
+
+Steps to convert a bot to a Skill using SDK v3.30.0:
+1)	In the v3 bot's message controller, change the attribute applied to the controller from `BotAuthentication` to `SkillBotAuthentication`.
+1)	In the bot's web.config file, add an `AllowedCallers` app setting. The allowed callers array can restrict which skill consumers can access the skill. Set this to "*" to accept calls from any skill consumer, or set it to a comma-delimited list of the app IDs of the allowed consumers.
+1)	Add code to send the skill host `EndOfConversation` when the skill is finished and to properly handle `EndOfConversation` from the skill consumer.
+
+(Note: these changes will not affect the bot's behavior as a normal v3 bot, and it can still be used that way, too.)
+
+
+## Minimum Prerequisites
+
+- [.NET Core SDK 2.1](https://dotnet.microsoft.com/download) 
+
+  ```bash
+  # determine dotnet versions installed
+  dotnet --list-sdks
+  ```
+
+- [.NET Framework 4.6.1](https://dotnet.microsoft.com/download/dotnet-framework) or greater
+
+  [How to: Determine which .NET Framework versions are installed](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed)
+
+## Key concepts in this sample
+
+The solution includes a parent bot (`SimpleRootBot`) built using [v4.7.2](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.7.2) and three skill bots (`EchoBot`, `PizzaBot`, `SimpleSandwichBot`) built using [v3.30.0](https://www.nuget.org/packages/Microsoft.Bot.Builder/3.30.0). This example demonstrates how a parent bot can post activities to skill bots and returns the skill responses to the user.
+
+- `SimpleRootBot`: this project is based on [80.skills-simple-bot-to-bot](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/80.skills-simple-bot-to-bot) It shows how to consume the three V3 skills and includes:
+  - A [RootBot](V4SimpleRootBot/Bots/RootBot.cs) which calls the user chosen skill and keeps the conversation active until the user says "end" or "stop". [RootBot](V4SimpleRootBot/Bots/RootBot.cs) also keeps track of the conversation with the skill and handles the `EndOfConversation` activity received from the skill to terminate the conversation.
+  - A simple [SkillConversationIdFactory](V4SimpleRootBot/SkillConversationIdFactory.cs) based on an in memory `ConcurrentDictionary` that creates and maintains conversation IDs used to interact with a skill
```

Comment:
For links to files in the sample code, use Markdown-style links with a relative URL. I'm not sure why these links are broken when I view the README in the PR, but I assume they'll resolve correctly once it's merged into master.

For links to v3 or v4 ref docs, use Markdown-style links with an absolute URL (minus the language code (en-us) portion of the URL).

Reply:
These links resolve using my markdown editor... not sure why they don't during GitHub pr review.